nodecg-io-youtube/extension/index.ts

```diff
@@ -0,0 +1,68 @@
+import { NodeCG } from "nodecg/types/server";
+import { ServiceProvider } from "nodecg-io-core/extension/types";
+import { emptySuccess, success, Result } from "nodecg-io-core/extension/utils/result";
+import { ServiceBundle } from "nodecg-io-core/extension/serviceBundle";
+import { youtube_v3, google } from "googleapis";
+import * as express from "express";
+import opn = require("open");
+
+interface YoutubeServiceConfig {
+    clientID: string;
+    clientSecret: string;
+}
+
+export interface YoutubeServiceClient {
+    getRawClient(): youtube_v3.Youtube;
+}
+
+module.exports = (nodecg: NodeCG): ServiceProvider<YoutubeServiceClient> | undefined => {
+    const youtubeService = new YoutubeService(nodecg, "youtube", __dirname, "../youtube-schema.json");
+    return youtubeService.register();
+};
+
+class YoutubeService extends ServiceBundle<YoutubeServiceConfig, YoutubeServiceClient> {
+    async validateConfig(_config: YoutubeServiceConfig): Promise<Result<void>> {
+        return emptySuccess();
+    }
+
+    async createClient(config: YoutubeServiceConfig): Promise<Result<YoutubeServiceClient>> {
+        // https://developers.google.com/identity/protocols/oauth2/web-server#creatingcred
+        const auth = new google.auth.OAuth2({
+            clientId: config.clientID,
+            clientSecret: config.clientSecret,
+            redirectUri: "localhost:9090/nodecg-io-youtube/oauth2callback",
```

Comment:
@LarsVomMars we need to prepend the protocol here, in this case http. I've tested it with http and authentication worked. I didn't test any functions of the client though.
```suggestion
            redirectUri: "http://localhost:9090/nodecg-io-youtube/oauth2callback",
```

Reply:
lol